### ISSUE TITLE
root-config --cflags returns -m32 instead of -m64

### ISSUE THREAD

**Issue by adamdddave** [conda-forge/root-feedstock#41]
Issue:
Trying to compile a script with root leads to the following:
```
In file included from /home/davis3a4/miniconda/envs/HEPenv/x86_64-conda_cos6-linux-gnu/sysroot/usr/include/features.h:380:0,
                 from /home/davis3a4/miniconda/envs/HEPenv/x86_64-conda_cos6-linux-gnu/sysroot/usr/include/stdio.h:28,
                 from makePDFs.cpp:3:
/home/davis3a4/miniconda/envs/HEPenv/x86_64-conda_cos6-linux-gnu/sysroot/usr/include/gnu/stubs.h:7:11: fatal error: gnu/stubs-32.h: No such file or directory
 # include <gnu/stubs-32.h>
           ^~~~~~~~~~~~~~~~
```
<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
# packages in environment at /home/davis3a4/miniconda/envs/HEPenv:
#
# Name                    Version                   Build  Channel
afterimage                1.21              hf014aae_1002    conda-forge
attrs                     19.1.0                     py_0    conda-forge
backcall                  0.1.0                      py_0    conda-forge
binutils                  1.0.1                         0    conda-forge
binutils_impl_linux-64    2.31.1               h6176602_1  
binutils_linux-64         2.31.1               h6176602_7  
bleach                    3.1.0                      py_0    conda-forge
bzip2                     1.0.6             h14c3975_1002    conda-forge
c-compiler                1.0.1                h14c3975_0    conda-forge
ca-certificates           2019.6.16            hecc5488_0    conda-forge
cairo                     1.16.0            h18b612c_1001    conda-forge
certifi                   2019.6.16                py37_0    conda-forge
cfitsio                   3.430             h0bd2449_1002    conda-forge
compilers                 1.0.1                         0    conda-forge
curl                      7.64.1               hf8cf82a_0    conda-forge
cxx-compiler              1.0.1                h6bb024c_0    conda-forge
davix                     0.7.3                hb54d6fb_2    conda-forge
dbus                      1.13.6               he372182_0    conda-forge
decorator                 4.4.0                      py_0    conda-forge
defusedxml                0.5.0                      py_1    conda-forge
entrypoints               0.3                   py37_1000    conda-forge
expat                     2.2.5             he1b5a44_1003    conda-forge
fftw                      3.3.8           mpi_mpich_h3f9e1be_1006    conda-forge
fontconfig                2.13.1            he4413a7_1000    conda-forge
fortran-compiler          1.0.1                h0e1e685_0    conda-forge
freetype                  2.10.0               he983fc9_0    conda-forge
gcc_impl_linux-64         7.3.0                habb00fd_1    conda-forge
gcc_linux-64              7.3.0                h553295d_7    conda-forge
gdk-pixbuf                2.36.12           h7a26e22_1003    conda-forge
gettext                   0.19.8.1          hc5be6a0_1002    conda-forge
gfortran_impl_linux-64    7.3.0                hdf63c60_1  
gfortran_linux-64         7.3.0                h553295d_7  
giflib                    5.1.7                h516909a_1    conda-forge
glew                      2.0.0             hf484d3e_1002    conda-forge
glib                      2.58.3            h6f030ca_1001    conda-forge
gobject-introspection     1.58.2          py37h5503ade_1001    conda-forge
graphite2                 1.3.13            hf484d3e_1000    conda-forge
graphviz                  2.40.1               h0dab3d1_0    conda-forge
gsl                       2.5                  h294904e_0    conda-forge
gst-plugins-base          1.14.5               h0935bb2_0    conda-forge
gstreamer                 1.14.5               h36ae1b5_0    conda-forge
gxx_impl_linux-64         7.3.0                hdf63c60_1    conda-forge
gxx_linux-64              7.3.0                h553295d_7    conda-forge
harfbuzz                  2.4.0                h37c48d4_1    conda-forge
icu                       58.2              hf484d3e_1000    conda-forge
ipykernel                 5.1.1            py37h24bf2e0_0    conda-forge
ipython                   7.5.0            py37h24bf2e0_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
jedi                      0.14.0                   py37_0    conda-forge
jinja2                    2.10.1                     py_0    conda-forge
jpeg                      9c                h14c3975_1001    conda-forge
jsonschema                3.0.1                    py37_0    conda-forge
jupyter_client            5.2.4                      py_3    conda-forge
jupyter_core              4.4.0                      py_0    conda-forge
krb5                      1.16.3            h05b26f9_1001    conda-forge
libblas                   3.8.0                7_openblas    conda-forge
libcblas                  3.8.0                7_openblas    conda-forge
libcroco                  0.6.13               h8d621e5_0    conda-forge
libcurl                   7.64.1               hda55be3_0    conda-forge
libedit                   3.1.20170329      hf8c457e_1001    conda-forge
libffi                    3.2.1             he1b5a44_1006    conda-forge
libgcc-ng                 9.1.0                hdf63c60_0  
libgfortran-ng            7.3.0                hdf63c60_0  
libglu                    9.0.0             hf484d3e_1000    conda-forge
libiconv                  1.15              h516909a_1005    conda-forge
liblapack                 3.8.0                7_openblas    conda-forge
libpng                    1.6.37               hed695b0_0    conda-forge
librsvg                   2.44.14              h11c8777_0    conda-forge
libsodium                 1.0.16            h14c3975_1001    conda-forge
libssh2                   1.8.2                h22169c7_2    conda-forge
libstdcxx-ng              9.1.0                hdf63c60_0  
libtiff                   4.0.10            h57b8799_1003    conda-forge
libtool                   2.4.6             h14c3975_1002    conda-forge
libuuid                   2.32.1            h14c3975_1000    conda-forge
libxcb                    1.13              h14c3975_1002    conda-forge
libxml2                   2.9.9                h13577e0_0    conda-forge
lz4                       2.1.9            py37hd79334b_0    conda-forge
lz4-c                     1.8.3             he1b5a44_1001    conda-forge
markupsafe                1.1.1            py37h14c3975_0    conda-forge
metakernel                0.24.2                     py_0    conda-forge
mistune                   0.8.4           py37h14c3975_1000    conda-forge
mpi                       1.0                       mpich    conda-forge
mpich                     3.2.1             hc99cbb1_1012    conda-forge
nbconvert                 5.5.0                      py_0    conda-forge
nbformat                  4.4.0                      py_1    conda-forge
ncurses                   6.1               hf484d3e_1002    conda-forge
notebook                  5.7.8                    py37_1    conda-forge
numpy                     1.16.4           py37h95a1406_0    conda-forge
openblas                  0.3.5             h9ac9557_1001    conda-forge
openssl                   1.1.1b               h14c3975_1    conda-forge
pandoc                    2.7.3                         0    conda-forge
pandocfilters             1.4.2                      py_1    conda-forge
pango                     1.40.14           he7ab937_1005    conda-forge
parso                     0.5.0                      py_0    conda-forge
pcre                      8.41              hf484d3e_1003    conda-forge
pexpect                   4.7.0                    py37_0    conda-forge
pickleshare               0.7.5                 py37_1000    conda-forge
pip                       19.1.1                   py37_0    conda-forge
pixman                    0.38.0            h516909a_1003    conda-forge
prometheus_client         0.7.1                      py_0    conda-forge
prompt_toolkit            2.0.9                      py_0    conda-forge
pthread-stubs             0.4               h14c3975_1001    conda-forge
ptyprocess                0.6.0                   py_1001    conda-forge
pygments                  2.4.2                      py_0    conda-forge
pyrsistent                0.15.2           py37h516909a_0    conda-forge
pythia8                   8.240            py37he1b5a44_2    conda-forge
python                    3.7.3                h5b0a415_0    conda-forge
python-dateutil           2.8.0                      py_0    conda-forge
pyzmq                     18.0.2           py37hc4ba49a_0    conda-forge
qt                        5.9.7                h52cfd70_2    conda-forge
readline                  7.0               hf8c457e_1001    conda-forge
root                      6.18.00         py37h4d53dcf_14    conda-forge
send2trash                1.5.0                      py_0    conda-forge
setuptools                41.0.1                   py37_0    conda-forge
six                       1.12.0                py37_1000    conda-forge
sqlite                    3.28.0               h8b20d00_0    conda-forge
tbb                       2019.7               hc9558a2_0    conda-forge
tbb-devel                 2019.7               hc9558a2_0    conda-forge
terminado                 0.8.2                    py37_0    conda-forge
testpath                  0.4.2                   py_1001    conda-forge
tk                        8.6.9             hed695b0_1002    conda-forge
tornado                   6.0.3            py37h516909a_0    conda-forge
traitlets                 4.3.2                 py37_1000    conda-forge
vdt                       0.4.3                hf484d3e_0    conda-forge
wcwidth                   0.1.7                      py_1    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.33.4                   py37_0    conda-forge
xorg-fixesproto           5.0               h14c3975_1002    conda-forge
xorg-kbproto              1.0.7             h14c3975_1002    conda-forge
xorg-libice               1.0.9             h516909a_1004    conda-forge
xorg-libsm                1.2.3             h84519dc_1000    conda-forge
xorg-libx11               1.6.7             h14c3975_1000    conda-forge
xorg-libxau               1.0.9                h14c3975_0    conda-forge
xorg-libxcursor           1.2.0                h14c3975_0    conda-forge
xorg-libxdmcp             1.1.3                h516909a_0    conda-forge
xorg-libxext              1.3.4                h516909a_0    conda-forge
xorg-libxfixes            5.0.3             h516909a_1004    conda-forge
xorg-libxft               2.3.3                h71203ad_0    conda-forge
xorg-libxpm               3.5.12            h14c3975_1002    conda-forge
xorg-libxrender           0.9.10            h516909a_1002    conda-forge
xorg-libxt                1.1.5             h516909a_1003    conda-forge
xorg-renderproto          0.11.1            h14c3975_1002    conda-forge
xorg-xextproto            7.3.0             h14c3975_1002    conda-forge
xorg-xproto               7.0.31            h14c3975_1007    conda-forge
xrootd                    4.9.1            py37h1b6aa1e_0    conda-forge
xz                        5.2.4             h14c3975_1001    conda-forge
zeromq                    4.3.1             hf484d3e_1000    conda-forge
zlib                      1.2.11            h14c3975_1004    conda-forge
zstd                      1.4.0                h3b9ef0a_0    conda-forge
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

     active environment : HEPenv
    active env location : /home/davis3a4/miniconda/envs/HEPenv
            shell level : 1
       user config file : /home/davis3a4/.condarc
 populated config files : 
          conda version : 4.7.5
    conda-build version : not installed
         python version : 3.7.3.final.0
       virtual packages : __cuda=10.1
       base environment : /home/davis3a4/miniconda  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/davis3a4/miniconda/pkgs
                          /home/davis3a4/.conda/pkgs
       envs directories : /home/davis3a4/miniconda/envs
                          /home/davis3a4/.conda/envs
               platform : linux-64
             user-agent : conda/4.7.5 requests/2.21.0 CPython/3.7.3 Linux/3.10.0-957.10.1.el7.x86_64 centos/7.6.1810 glibc/2.17
                UID:GID : 1005:1005
             netrc file : None
           offline mode : False

```
</details>

**Comment by chrisburr** [comment]
I think you're trying to compile a 32-bit executable but conda-forge only provides 64-bit binaries. Do you have `-m32` somewhere in your compile command? If so, you should replace it with `-m64`.

If not, can you share your source and how you're compiling it? (privately if needed)

**Comment by adamdddave** [comment]
I don't have `-m32` defined explicitly, but I think it comes from `$ROOTSYS/bin/root-config --cflags` which returns `-pthread -std=c++1z -m32 -msse -mfpmath=sse -I/home/davis3a4/miniconda/envs/HEPenv/include`

**Comment by chrisburr** [comment]
Ah that should be fixed, I'll take a look soon.

For now you can work around it by adding `-m64` after the output of `root-config` or you can patch it in place with:
```bash
sed -i.bak 's/-m32/-m64/g' $(which root-config)
```
(Provided you only run it once, this will create a backup of the original file called `root-config.bak`)

**Comment by chrisburr** [comment]
Fixed builds of 6.18.00 should be available in a couple of hours.

Thanks for reporting! 🍰
